identity-account/src/account/account.rs

```diff
@@ -367,16 +428,26 @@ impl Account {
 
     let mut diff: DiffMessage = DiffMessage::new(old_doc, new_doc, *previous_message_id)?;
 
-    let method: &IotaVerificationMethod = old_state.document().default_signing_method()?;
+    let signing_method: &IotaVerificationMethod = match signing_method_query {
+      Some(fragment) => old_state
+        .document()
+        .resolve_method_with_scope(fragment, MethodScope::capability_invocation())
+        .ok_or(Error::DIDError(identity_did::Error::MethodNotFound))?,
+      None => old_state.document().default_signing_method()?,
+    };
 
```

Comment:
This should use `is_signing_method` too since `sign_data` bypasses the check added in `sign_self`.